File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        def update(m, n):
            nonlocal msize
            for r in range(max(0, m - 2), min(m + 1, msize)):
                for c in range(max(0, n - 2), min(n + 1, msize)):
                    r1, c1 = r % (size - 2), c % (size - 2)
                    max_local[r1][c1] = max(grid[m][n], max_local[r1][c1])
        
        size = len(grid)
        msize = size - 2
        max_local = [([0] * msize) for _ in range(msize)]
        for i in range(size):
            for j in range(size):
                update(i, j)
        
        return max_local

File: 2373-largest-local-values-in-a-matrix/2373-largest-local-values-in-a-matrix.go
func largestLocal(grid [][]int) [][]int {
    n := len(grid)
    res := [][]int{}
    for range n - 2 {
        res = append(res, make([]int, n - 2))
    }
    
    for r := 1; r < n - 1; r++ {
        for c := 1; c < n - 1; c++ {
            res[r - 1][c - 1] = grid[r][c]
            for _, dir := range [][]int{{1,0}, {0,1}, {-1,0}, {0,-1},
                                     {-1,-1}, {1,-1}, {-1,1}, {1,1}} {
                
                res[r - 1][c - 1] = max(res[r - 1][c - 1], 
                                        grid[r + dir[0]][c + dir[1]])
            }
        }
    }
    return res
}